test/built-ins/AsyncGeneratorPrototype/next/this-val-not-object.js

```diff
@@ -0,0 +1,98 @@
+// Copyright 2018 Valerie Young. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-asyncgenerator-prototype-next
+description: next rejects promise when `this` value not an object
+info: |
+  AsyncGenerator.prototype.next ( value )
+  1. Let generator be the this value.
+  2. Let completion be NormalCompletion(value).
+  3. Return ! AsyncGeneratorEnqueue(generator, completion).
+
+  AsyncGeneratorEnqueue ( generator, completion )
+  ...
+  3. If Type(generator) is not Object, or if generator does not have an
+     [[AsyncGeneratorState]] internal slot, then
+    a. Let badGeneratorError be a newly created TypeError object.
+    b. Perform ! Call(promiseCapability.[[Reject]], undefined, « badGeneratorError »).
+    c. Return promiseCapability.[[Promise]].
+
+flags: [async]
+features: [async-iteration]
```

Comment:
I'd say add `Symbol` to the `features:` list of any test files that include test code that contains `Symbol`, but I'd actually like you to _not_ add them in, because I want to use this PR to try out a tool I'm working on (just in case anyone else asks for them to be added)